frontend/workflows/k8s/src/scale-resources.tsx

```diff
@@ -0,0 +1,286 @@
+              validation: string().matches(QUANTITY_REGEX),
+            },
+          },
+          {
+            name: "Memory Limit",
+            value: deployment.deploymentSpec.template.spec.containers.find(
+              container => container.name === containerName
+            ).resources.limits.memory,
+            textFieldLabels: {
+              disabledField: "Current Limit",
+              updatedField: "New limit",
+            },
+            input: {
+              type: "string",
+              key: `deploymentSpec.template.spec.containers[${containerIndex}].resources.limits.memory`,
+              validation: string().matches(QUANTITY_REGEX),
+            },
+          },
+          {
+            name: "Memory Request",
+            value: deployment.deploymentSpec.template.spec.containers.find(
+              container => container.name === containerName
+            ).resources.requests.memory,
+            textFieldLabels: {
+              disabledField: "Current Request",
+              updatedField: "New Request",
+            },
+            input: {
+              type: "string",
+              key: `deploymentSpec.template.spec.containers[${containerIndex}].resources.requests.memory`,
+              validation: string().matches(QUANTITY_REGEX),
+            },
+          },
+        ]}
+      />
+      <ButtonGroup>
+        <Button text="Back" variant="neutral" onClick={() => onBack()} />
+        <Button text="Update" variant="destructive" onClick={onSubmit} />
+      </ButtonGroup>
+    </WizardStep>
+  );
+};
+
+function fortmatResourceString(resourceName: string, resourceRequirement: string): string {
+  // Capitalize the first letter of resourceName
+  const capitalizedResourceName = resourceName.charAt(0).toUpperCase() + resourceName.slice(1);
+
+  // Capitalize and remove the s at the end of resourceRequirement
+  const modifiedResourceRequirement =
+    resourceRequirement.charAt(0).toUpperCase() + resourceRequirement.slice(1, -1);
+
+  // Return the modified strings
+  return `${capitalizedResourceName} ${modifiedResourceRequirement}`;
+}
+
+const Confirm: React.FC<ConfirmChild> = () => {
+  const deployment = useDataLayout("deploymentData").displayValue() as IClutch.k8s.v1.Deployment;
+  const updateData = useDataLayout("updateData");
+  const currentDeploymentData = useDataLayout(
+    "currentDeploymentData"
+  ).displayValue() as IClutch.k8s.v1.Deployment;
+
+  const updateRows: any[] = [];
+
+  let updatedContainer = false;
+  deployment.deploymentSpec.template.spec.containers.forEach(container => {
+    Object.keys(container.resources).forEach(resourceRequirement => {
+      Object.keys(container.resources[resourceRequirement]).forEach(resourceName => {
+        const newValue = container.resources[resourceRequirement][resourceName];
+        const oldValue = currentDeploymentData.deploymentSpec.template.spec.containers.find(
+          oldContainer => oldContainer.name === container.name
```

Comment:
i'd probably make a helper function for getting the correct container since we're repeating this 4 or 5 times. wdyt?

Reply:
I went ahead and implemented a helper function. I think it helps make the `ScaleResources` component more readable at least.